frame/node-authorization/src/lib.rs

```diff
@@ -0,0 +1,761 @@
+        #[weight = T::WeightInfo::add_connections()]
+        pub fn add_connections(
+            origin,
+            node: NodePublicKey,
+            connections: Vec<NodePublicKey>
+        ) {
+            let sender = ensure_signed(origin)?;
+            ensure!(Owners::<T>::contains_key(&node), Error::<T>::NotClaimed);
+            ensure!(Owners::<T>::get(&node) == sender, Error::<T>::NotOwner);
+
+            let mut nodes = AdditionalConnections::get(&node);
+
+            for add_node in connections.iter() {
+                if *add_node == node {
+                    continue;
+                }
+                if let Err(add_location) = nodes.binary_search(add_node) {
+                    nodes.insert(add_location, add_node.clone());
+                }
+            }
+            
+            AdditionalConnections::insert(&node, nodes);
+
+            Self::deposit_event(RawEvent::ConnectionsAdded(node, connections));
+        }
+
+        /// Remove additional connections of a given node.
+        ///
+        /// - `node`: identifier of the node.
+        /// - `connections`: additonal nodes from which the connections are not allowed anymore.
+        #[weight = T::WeightInfo::remove_connections()]
+        pub fn remove_connections(
+            origin,
+            node: NodePublicKey,
+            connections: Vec<NodePublicKey>
+        ) {
+            let sender = ensure_signed(origin)?;
+            ensure!(Owners::<T>::contains_key(&node), Error::<T>::NotClaimed);
+            ensure!(Owners::<T>::get(&node) == sender, Error::<T>::NotOwner);
+
+            let mut nodes = AdditionalConnections::get(&node);
+
+            for remove_node in connections.iter() {
+                if let Ok(remove_location) = nodes.binary_search(remove_node) {
+                    nodes.remove(remove_location);
+                }
+            }
+            
+            AdditionalConnections::insert(&node, nodes);
+
+            Self::deposit_event(RawEvent::ConnectionsRemoved(node, connections));
+        }
+
+        /// Set reserved node every block. If may not be enabled depends on the offchain
+        /// worker CLI flag.
+        fn offchain_worker(now: T::BlockNumber) {
+            let node_public_key = sp_io::offchain::get_node_public_key();
+            match node_public_key {
+                Err(_) => debug::error!("Error: failed to get public key of node at {:?}", now),
+                Ok(node) => sp_io::offchain::set_reserved_nodes(
+                    Self::get_authorized_nodes(&node), true
+                ),
+            }
+        }
+    }
+}
+
+impl<T: Trait> Module<T> {
+    fn initialize_nodes(nodes: &Vec<(NodePublicKey, T::AccountId)>) {
+        let mut node_public_keys = nodes.iter()
+            .map(|item| item.0.clone())
+            .collect::<Vec<NodePublicKey>>();
+        node_public_keys.sort();
+        WellKnownNodes::put(&node_public_keys);
```

Comment:
we should reset `AdditionalConnections` here as well, cause otherwise it may contain some stray entries.

Reply:
I guess we want to keep `Owners/AdditionalConnections` detached from `WellKnownNodes` so I suppose it's fine as-is.